src/python/pants/backend/experimental/python/typecheck/pyright/register.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+"""Static type checker for Python, running on NodeJS.
+
+See https://github.com/Microsoft/pyright for details.
+"""
+
+from __future__ import annotations
+
+from typing import Iterable
+
+from pants.backend.javascript.subsystems import nodejs
+from pants.backend.python.typecheck.pyright import rules as pyright_rules
+from pants.backend.python.typecheck.pyright import skip_field
+from pants.engine.rules import Rule
+from pants.engine.unions import UnionRule
+
+
+def rules() -> Iterable[Rule | UnionRule]:
+    return (
+        *nodejs.rules(),
```

Comment:
@Eric-Arellano Per Slack: Would I add the `target_types` in the rules?

Reply:
Ohhh, I thought that was only to define/declare NEW targets. Didn't think it was also to notify the system about existing deps. 

I think because a lot of the docs refer to "custom" targets when that function comes up: https://www.pantsbuild.org/docs/target-api-new-targets#step-2-register-the-target-type-in-registerpy

Reply:
It basically means "make sure these targets can be used!!" It's fine for multiple backends to specify the same value multiple times. We dedupe